LayoutTests/fast/forms/number/number-validity-badinput.html

```diff
@@ -53,7 +53,13 @@
 // Visible value is '-1a'.
 number.focus();
 document.execCommand('SelectAll');
-shouldBeEqualToString('document.getSelection().toString()', '-1a');
+function visibleValue(input) {
+    const shadowRoot = internals.shadowRoot(input);
```

Comment:
Too bad we can’t do this test without the internals, but I assume you would have done it another way if possible.

Reply:
Yeah, there isn't really a way to get the visible text out of input element.